src/runtime_src/core/common/memaccess.cpp

```diff
@@ -0,0 +1,217 @@
+/**
+ * Copyright (C) 2016-2021 Xilinx, Inc
+ * Copyright (C) 2022 Advanced Micro Devices, Inc. - All rights reserved
+ * 
+ * Author: Sonal Santan
+ * Simple command line utility to interact with PCIe devices
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may
+ * not use this file except in compliance with the License. A copy of the
+ * License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+#define XRT_CORE_COMMON_SOURCE
+#include "memaccess.h"
+
+#include <iostream>
+#include <numeric>
+#include <vector>
+
+#include "memalign.h"
+#include "query_requests.h"
+#include "utils.h"
+#include "core/common/unistd.h"
+
+namespace {
+// Holds the parsed data from the memory topology object
+struct mem_bank_t
+{
+  uint64_t m_base_address;
+  uint64_t m_size;
+  mem_bank_t(const struct mem_data& data) :
+  m_base_address(data.m_base_address)
+  // In memory topology struct size is stored as KB. We convert to bytes for easy referencing
```

Comment:
Please add the comment on the same line as the initialization of the m_size variable, as well as, indent the variables being initialized.  This will greatly help future developers reading this code.